code/ecdsap256/Hacl.P256.fsti

```diff
@@ -1,501 +1,345 @@
+    (b <==> Some? (S.pk_uncompressed_to_raw (as_seq h0 pk))) /\
+    (b ==> (as_seq h1 pk_raw == Some?.v (S.pk_uncompressed_to_raw (as_seq h0 pk)))))
+
+
+[@@ Comment "Convert a public key from compressed to its raw form.
+
+  The function returns `true` for successful conversion of a public key and `false` otherwise.
+
+  The outparam `pk_raw` points to 64 bytes of valid memory, i.e., uint8_t[64].
+  The argument `pk` points to 33 bytes of valid memory, i.e., uint8_t[33].
+
+  The function also checks whether (x, y) is a valid point."]
+val compressed_to_raw: pk:lbuffer uint8 33ul -> pk_raw:lbuffer uint8 64ul -> Stack bool
+  (requires fun h -> live h pk /\ live h pk_raw /\ disjoint pk pk_raw)
+  (ensures  fun h0 b h1 -> modifies (loc pk_raw) h0 h1 /\
+    (b <==> Some? (S.pk_compressed_to_raw (as_seq h0 pk))) /\
+    (b ==> (as_seq h1 pk_raw == Some?.v (S.pk_compressed_to_raw (as_seq h0 pk)))))
+
+
+[@@ Comment "Convert a public key from raw to its uncompressed form.
+
+  The outparam `pk` points to 65 bytes of valid memory, i.e., uint8_t[65].
+  The argument `pk_raw` points to 64 bytes of valid memory, i.e., uint8_t[64].
+
+  The function DOESN'T check whether (x, y) is a valid point."]
+val raw_to_uncompressed: pk_raw:lbuffer uint8 64ul -> pk:lbuffer uint8 65ul -> Stack unit
+  (requires fun h -> live h pk /\ live h pk_raw /\ disjoint pk pk_raw)
+  (ensures fun h0 _ h1 -> modifies (loc pk) h0 h1 /\
+    as_seq h1 pk == S.pk_uncompressed_from_raw (as_seq h0 pk_raw))
+
+
+[@@ Comment "Convert a public key from raw to its compressed form.
+
+  The outparam `pk` points to 33 bytes of valid memory, i.e., uint8_t[33].
+  The argument `pk_raw` points to 64 bytes of valid memory, i.e., uint8_t[64].
+
+  The function DOESN'T check whether (x, y) is a valid point."]
+val raw_to_compressed: pk_raw:lbuffer uint8 64ul -> pk:lbuffer uint8 33ul -> Stack unit
+  (requires fun h -> live h pk /\ live h pk_raw /\ disjoint pk pk_raw)
+  (ensures  fun h0 _ h1 -> modifies (loc pk) h0 h1 /\
+    as_seq h1 pk == S.pk_compressed_from_raw (as_seq h0 pk_raw))
 
 
 [@@ CPrologue "
 /******************/
 /* ECDH agreement */
 /******************/";
 
-(Comment "Convert a private key into a raw public key.
+Comment "Compute the public key from the private key.
 
-This function performs no key validation.
+  The function returns `true` if a private key is valid and `false` otherwise.
 
-  Input: `scalar`, the private key, of type `uint8[32]`.
-  Output: `result`, the public key, of type `uint8[64]`.
-  Returns:
-  - `true`, for success, meaning the public key is not a point at infinity
-  - `false`, otherwise.
+  The outparam `public_key`  points to 64 bytes of valid memory, i.e., uint8_t[64].
+  The argument `private_key` points to 32 bytes of valid memory, i.e., uint8_t[32].
 
-  `scalar` and `result` MUST NOT overlap.")]
+  The private key is valid:
+    • 0 < `private_key` < the order of the curve."]
 val dh_initiator:
```

Comment:
This naming does not make much sense: this should be secret_to_public maybe?

Reply:
will be done in a separate PR